app/server/lib/requestUtils.ts

```diff
@@ -102,15 +103,26 @@ export function allowHost(req: Request, allowedHost: string|URL) {
   const allowedUrl = (typeof allowedHost === 'string') ? new URL(`${proto}://${allowedHost}`) : allowedHost;
   if (mreq.isCustomHost) {
     // For a request to a custom domain, the full hostname must match.
-    return actualUrl.hostname === allowedUrl.hostname;
+  return actualUrl.hostname === allowedUrl.hostname;
   } else {
     // For requests to a native subdomains, only the base domain needs to match.
     const allowedDomain = parseSubdomain(allowedUrl.hostname);
     const actualDomain = parseSubdomain(actualUrl.hostname);
-    return (actualDomain.base === allowedDomain.base);
+  return (!_.isEmpty(actualDomain) ? actualDomain.base === allowedDomain.base : allowedUrl.hostname === actualUrl.hostname);
```

Comment:
Here I had to make some change because for url like `https://badexample.com` we would have
```js
const allowedDomain = parseSubdomain(allowedUrl.hostname)  // = {}
const actualDomain = parseSubdomain(actualUrl.hostname);   // = {}
return actualDomain.base === allowedDomain.base            // ==> undefined === undefined ==> true
```